docs/architecture/structure.md

```diff
@@ -0,0 +1,106 @@
+# Structure of the ETL
+
+## A compute graph
+
+The ETL is a compute graph, that uses a directed acyclic graph (DAG) to describe the dependencies between datasets. There are currently three types of nodes in the graph: snapshots, datasets and grapher exports.
+
+### Snapshots
+
+Data snapshots represent a copy of an upstream datasource in its original format, at a particular point in time.
+
+New-style snapshots are managed in the ETL using DVC and begin with the prefix `snapshot://`. Old-style snapshots are managed by the [walden](https://github.com/owid/walden) codebase and begin with the prefix `walden://`.
+
+Snapshot URIs use the convention: 
+
+```
+snapshot://<provider>/<version>/<filename>
+```
+
+Examples:
+
+- `snapshot://aviation_safety_network/2022-10-14/aviation_statistics.csv`
+- `walden://irena/2022-10-07/renewable_power_generation_costs`
+
+### Datasets
+
+Datasets are nodes in the compute graph, the main units of work in the ETL. They represent a transformation from one or more ingredients into a better, more useful output.
+
+Dataset URIs use the convention:
+
+```
+data://<channel>/<provider>/<version>/<dataset>
+```
+
+If the dataset is part of a large topic that involves many providers, the name of the topic can be used instead of a provider name.
+
+The dataset URI is used to identify the code that builds the dataset, and also to identify the output file on disk.
+
+Example:
+
+- URI: `data://garden/faostat/2022-05-17/faostat_fa`
+- Code: `etl/steps/data/garden/faostat/2022-05-17/faostat_fa.py`
+- Dataset prefix: `s3://owid-catalog/garden/faostat/2022-05-17/faostat_fa`
+
+### Grapher views
+
+The Grapher codebase can only except datasets that are in a particulare shape:
+
+```
+(entity, year, variable, value)
+```
+
+However, datasets in the ETL are often in a different shape. For example, they may be broken down by gender, disease type, fish stock, or some other dimension.
+
+All data charted on our site is built from simplified grapher views, stored in MySQL. 
+Grapher steps are responsible for reshaping a dataset on disk into a grapher view, and then uploading it to MySQL. A single variable may fan out into a large number of grapher views.
+
+Grapher views use the following convention:
+
+```
+data://grapher/<provider>/<version>/<dataset>
+```
+
+??? How do automatic grapher steps get created from here?
```

Comment:
It's black magic, but here are some notable things we do:
- we create variable for every combination of dimensions (e.g. `Death rate (sex=male, age group=60-70)`)
- country are converted to `entity_ids` (and new are created if they don't exist in our DB)
- in case we have multiple sources, we merge them into one (as grapher only supports one source)

Reply:
Gold! Thanks, I added these things in.